### PR TITLE
Feat : JRD : Change the start time of the Camel ingestion process to past midnight

### PR DESCRIPTION
**Pull request by souravroy2021**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-3345

### Change description ###

Change the start time of the Camel ingestion process to past midnight

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by SujeethKumarAri — APPROVED** [review]

**Review by kavithagade12 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/FileReadProcessor.java [hunk 1]
```diff
@@ -81,11 +86,11 @@ public void process(Exchange exchange) {
         String schedulerTime = context.getGlobalOption(SCHEDULER_START_TIME);
 
         //Check Stale OR Not existing file and exit run with proper error message
-        if (getFileStatusInBlobContainer(fileName, schedulerTime).equals(STALE)) {
+        if (getFileStatusInBlobContainer(routeProperties, schedulerTime).equals(STALE)) {
             exchange.getMessage().setHeader(IS_FILE_STALE, true);
             throw new RouteFailedException(String.format(STALE_FILE_ERROR,
                 fileName, DateFormatUtils.format(fileTimeStamp, "yyyy-MM-dd HH:mm:SS")));
```
Comment (ManishParyani) [inline src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/FileReadProcessor.java:92]: can we use routeProperties.getName at this place(L92) as well?

File: src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/FileReadProcessor.java [hunk 2]
```diff
@@ -119,7 +125,19 @@ private BlobStatus getFileStatusInBlobContainer(String fileName, String schedule
             if (cloudBlockBlob.exists()) {
                 cloudBlockBlob.downloadAttributes();
                 fileTimeStamp = cloudBlockBlob.getProperties().getLastModified();
-                return (isSameDay(fileTimeStamp, new Date(Long.valueOf(schedulerTime)))) ? NEW : STALE;
+
+                BlobStatus blobStatus;
+                Date today = new Date(Long.parseLong(schedulerTime));
```
Comment (ManishParyani) [inline src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/FileReadProcessor.java:130]: Can't we use LocalDate?

Comment (souravroy2021) [reply src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/FileReadProcessor.java:130]: It should be consistent, it needs refactoring to replace Date with LocalDate, which is not in scope of this ticket.
To keep it backward compatible I've used Date.

File: src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/FileReadProcessor.java (hunk 1 above)
Comment (souravroy2021) [reply src/main/java/uk/gov/hmcts/reform/data/ingestion/camel/processor/FileReadProcessor.java:92]: It should be part of refactoring as it belongs to existing code and not in scope of this ticket. Line 92 & 95 should be consistent, I agree.
Code duplication should be removed.

**Review by ManishParyani — APPROVED** [review]

**Pull request merged**